zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -121,6 +123,29 @@ public static Either<ProblemDetail, ActivateJobsRequest> toJobsActivationRequest
                             activationRequest, JobActivationRequest::getRequestTimeout))));
   }
 
+  public static Either<ProblemDetail, FailJobRequest> toJobFailRequest(
+      final JobFailRequest failRequest, final long jobKey) {
+
+    final var validationErrorResponse = validateJobFailRequest(failRequest);
+    return validationErrorResponse
+        .<Either<ProblemDetail, FailJobRequest>>map(Either::left)
+        .orElseGet(
+            () ->
+                Either.right(
+                    new FailJobRequest(
+                        jobKey,
+                        getIntOrDefault(
+                            failRequest,
+                            JobFailRequest::getRetries,
+                            JobServices.DEFAULT_RETRIES_VALUE),
+                        getStringOrEmpty(failRequest, JobFailRequest::getErrorMessage),
+                        getLongOrDefault(
+                            failRequest,
+                            JobFailRequest::getRetryBackOff,
+                            JobServices.DEFAULT_RETRYBACKOFF_VALUE),
```

Comment:
🔧 Let's just go for `getIntOrZero` and `getLongOrZero`. Zero will be the defacto default value for numeric values, just like for the gRPC API. We can add explicit defaults to the YAML and ensure here it's adhered to even if a default is missing in the YAML by accident.